Update supported versions and other upgrade documentation (#15588)
decidim/decidim: docs/modules/install/pages/update.adoc

[source,ruby]
----
gem "decidim", "0.29.0"
gem "decidim-conferences", "0.29.0"
gem "decidim-initiatives", "0.29.0"

gem "decidim-dev", "0.29.0"
----

For example, if the latest release is 0.29 we could decide to update to 0.30 or even 0.31.

> 

Reviewer: This phrase doesn't make sense anymore and it's much better explained below. 
```suggestion
```

> To update, usually we recommend using small increments, as from now and then, we are changing Ruby version, or we perform upgrades to our dependencies, work that could break your application if you do major leaps. Therefore, if you need to upgrade from 0.29 to 0.31, we recommend to first upgrade to 0.30, follow the release notes, then, from 0.30 to upgrade to 0.31.

Reviewer: ```suggestion
To update, we recommend using small increments. From time to time we change the Ruby version or update dependencies, and skipping multiple versions at once can introduce breaking changes. Therefore, if you need to upgrade from v0.29.0 to v0.31.0, we recommend to first upgrade to v0.30.0, follow the release notes, and then, from v0.30.0 to upgrade to v0.31.0.
```

> So, to perform the upgrade, we first follow the https://github.com/decidim/decidim/releases/tag/v0.30.0[release notes for 0.30], then, after we confirmed everything went ok, we should follow the https://github.com/decidim/decidim/releases/tag/v0.31.0[release notes for 0.31].

Reviewer: ```suggestion
So, to perform the upgrade, we first follow the https://github.com/decidim/decidim/releases/tag/v0.30.0[release notes for v0.30.0], then, after we confirmed everything went OK, we should follow the https://github.com/decidim/decidim/releases/tag/v0.31.0[release notes for v0.31.0].
```

== Recommendations

. Make a full backup of the database before updating, just in case something unexpected happens.

> . Never add 2 or more versions of Decidim to the same release, as the database structure could change, deprecated functions could be removed, causing issues that may prevent you to successfully run post release commands.

Reviewer: What do you mean with release here? Like "deployment to production"?? It is confusing because we're talking about releases of Decidim here too

Author: What i mean here is not to try to deploy an upgrade having 2 decidim versions ... will reprphase

Author: done in [1b48d7b](https://github.com/decidim/decidim/pull/15588/commits/1b48d7b82aa84331d1c9a4058e44309dc2d2c747)
